tools/wptrunner/wptrunner/browsers/firefox_android.py

```diff
@@ -87,14 +90,45 @@ def env_extras(**kwargs):
     return []
 
 
-def run_info_extras(**kwargs):
-    rv = fx_run_info_extras(**kwargs)
+def run_info_extras(logger, **kwargs):
+    rv = fx_run_info_extras(logger, **kwargs)
     package = kwargs["package_name"]
     rv.update({"e10s": True if package is not None and "geckoview" in package else False,
                "headless": False})
+
+    if kwargs["browser_version"] is None:
+        rv.update(run_info_browser_version(**kwargs))
+
+        if rv.get("browser_version") is None:
+            # If we didn't get the browser version from the apk, try to get it from adb dumpsys
+            rv["browser_version"] = get_package_browser_version(logger,
+                                                                kwargs["adb_binary"],
+                                                                kwargs["package_name"])
+
     return rv
 
 
+def get_package_browser_version(logger, adb_binary, package_name):
```

Comment:
Is there a reason not to use the same logic from here?

https://github.com/web-platform-tests/wpt/blob/d153eb36f37e71f3b64017952d6cc6d88724ed54/tools/wpt/browser.py#L1452-L1471

I imagine there could be a new abstract class called AndroidBase `class AndroidBase(BrowserSetup)`. AndroidBase has the `version` method implemented (by moving the implementation from ChromeAndroidBase to AndroidBase) .

Both ChromeAndroidBase and FirefoxAndroid are child classes of this new class. 

That way we don't have to repeat the same code or have a different pattern for getting the version.

Reply:
Yes, so I don't want this in the wpt frontend because at least gecko CI doesn't use that frontend, and I'd prefer to keep this logic in the shared code, as it is for desktop Firefox.

Arguably we could move stuff out of the frontend for other browsers, but that seemed out of scope for this PR.